File: src/main/java/org/akkreditierung/ui/page/StudiengangDetailPage.java
package org.akkreditierung.ui.page;

import org.akkreditierung.ui.model.*;
import org.apache.wicket.MarkupContainer;
import org.apache.wicket.ajax.AjaxRequestTarget;
import org.apache.wicket.ajax.form.AjaxFormComponentUpdatingBehavior;
import org.apache.wicket.ajax.markup.html.AjaxLink;
import org.apache.wicket.extensions.markup.html.repeater.data.grid.ICellPopulator;
import org.apache.wicket.extensions.markup.html.repeater.data.table.DefaultDataTable;
import org.apache.wicket.extensions.markup.html.repeater.data.table.IColumn;
import org.apache.wicket.extensions.markup.html.repeater.data.table.PropertyColumn;
import org.apache.wicket.markup.html.WebPage;
import org.apache.wicket.markup.html.basic.Label;
import org.apache.wicket.markup.html.form.TextField;
import org.apache.wicket.markup.html.link.BookmarkablePageLink;
import org.apache.wicket.markup.html.panel.Panel;
import org.apache.wicket.markup.repeater.Item;
import org.apache.wicket.markup.repeater.data.DataView;
import org.apache.wicket.markup.repeater.data.ListDataProvider;
import org.apache.wicket.model.IModel;
import org.apache.wicket.model.Model;
import org.apache.wicket.model.PropertyModel;
import org.apache.wicket.request.mapper.parameter.PageParameters;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Map;

public class StudiengangDetailPage extends WebPage {
    public static final String PAGE_PARAMETER_ID = "studiengangId";
	private static final long serialVersionUID = 1L;
    private AdvertiserConfig selected;

	public StudiengangDetailPage(final PageParameters parameters) {
		super(parameters);
        Map<String, StudiengaengeAttribute> studiengaengeAttributes = new StudiengaengeAttributeBean().findAll(parameters.get(PAGE_PARAMETER_ID).toInt(1));
        ListDataProvider<StudiengaengeAttribute> provider = new ListDataProvider<StudiengaengeAttribute>(new ArrayList(studiengaengeAttributes.values()));
        DataView<StudiengaengeAttribute> dataView = new DataView<StudiengaengeAttribute>("displayPanel", provider) {
            @Override
            protected void populateItem(Item<StudiengaengeAttribute> item) {
                StudiengaengeAttribute entry = item.getModelObject();
                item.add(new Label("key_column", entry.getK()));
                item.add(new Label("value_column", entry.getV()));
            }
        };
        add(dataView);
	}
}

File: src/main/scala/org/akkreditierung/ui/page/StudiengangDetailPage.scala
package org.akkreditierung.ui.page

import org.apache.wicket.markup.html.WebPage
import org.apache.wicket.markup.html.basic.Label
import org.apache.wicket.markup.repeater.Item
import org.apache.wicket.markup.repeater.data.ListDataProvider
import org.apache.wicket.request.mapper.parameter.PageParameters
import org.wicket.scala.RepeatingViews._
import org.wicket.scala.Fields._
import org.akkreditierung.model.DB
import org.akkreditierung.model.slick._
import scala.slick.jdbc.JdbcBackend.Database.dynamicSession
import collection.JavaConversions._

object StudiengangDetailPage {
  final val PAGE_PARAMETER_ID: String = "studiengangId"
  private final val serialVersionUID: Long = 1L
}

class StudiengangDetailPage(parameters: PageParameters) extends WebPage(parameters) {
  import DB.dal.profile.simple._
  val studiengaengeAttributes =  DB.db withDynSession DB.dal.studiengangAttributes.filter(_.id === getStudiengangId).list
  val provider = new ListDataProvider[StudiengangAttribute](studiengaengeAttributes)
  val d = dataView("displayPanel", provider) {(entry: StudiengangAttribute, item: Item[StudiengangAttribute]) =>
    item.add(new Label("key_column", entry.key))
    item.add(labelWithSpecialEscaping("value_column", entry.value))
  }
  add(d)

  def getStudiengangId = parameters.get(StudiengangDetailPage.PAGE_PARAMETER_ID).toInt(1)
}
